### DISCUSSION TITLE
detect objects in cropped images of a bigger image

### DISCUSSION THREAD

**Discussion by bit-scientist** [obss/sahi#638] [Q&A] [answered]
I have some questions before I use SAHI. Could you help me understand the method in more detail?

Let's say I have an [image ](https://www.freepik.com/free-photos-vectors/cat-background) with 3840x2160 size and I cropped it into 9 regions which makes each tile image's size 1280x720. (The image below is just an example )
![image](https://user-images.githubusercontent.com/26497571/190379482-b004d291-4cb8-4998-a538-0adeb3ac2794.png)

Then I annotated the 1280x720 image with bbox to train with yolov5m. Since the cropped image's size is 1280x720, I chose `python path/to/train.py --data cat_dataset.yaml --weights yolov5s.pt --img 1280` for training. 

For inference, I first send the original image (3840x2160) to a cropping function to prepare 9 crops, then send those 9 to the trained model. Objects in each smaller crops generate bboxes, but I need to draw the whole bbox on the whole image. The issue emerges here, I don't know how to merge individual bboxes of each crop into one bigger bbox. 

So, the question:
1. If I use SAHI,  with what resolution should I train the dataset, 3840x2160 or 1280x720?

I decided to crop images due to the following reasons:
- objects are very small (1-2%) compared to original image;
- objects are cluttered with background, so it's easier to crop and then annotate smaller images where objects appear;
- 4K image training doesn't seem right when considering all yolo models were trained with up to 1280 img-size.

I hope you can guide me correctly.
Thank you for the great work.

**Comment by fcakyon** [comment, marked as answer]
Thanks for your interest in SAHI!

You should train the model with 1280-sized slices. Then you should use slice height, width as 720, 1280 and `image_size=1280` in SAHI.

If you want no overlap between slices, you should use `overlap_height_ratio=0` and `overlap_width_ratio=0`.

Then SAHI will slice the original image (4k) into 720x1280 slices, perform yolov5 inference with 1280 input size for each slice, then map back the predictions into original image size coordinates (4k).

**Comment by bit-scientist** [comment]
Amazing. Thank you a lot, @fcakyon

**Reply by bit-scientist** [reply]
hi there @fcakyon , may I ask a follow-up question? 
Since the object sizes are very small compared to image size, I decided to crop the whole image into 9 crops above. Now I am thinking of cropping the image into 36 crops and use `imgsize 640` for training. Which of the two do you think will give better results in terms of both accuracy and speed, 9 crops per image with 1280 size or 36 crops per image with 640? I hope I explained it well.
